elements/lisk-p2p/src/directory/tried_peers.ts

```diff
@@ -0,0 +1,261 @@
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { P2PDiscoveredPeerInfo, P2PPeerInfo } from '../p2p_types';
+import { constructPeerIdFromPeerInfo, getBucket, PEER_TYPE } from '../utils';
+
+export const DEFAULT_TRIED_PEER_LIST_SIZE = 64;
+export const DEFAULT_TRIED_PEER_BUCKET_SIZE = 32;
+export const DEFAULT_MAX_RECONNECT_TRIES = 3;
+
+export interface TriedPeerConfig {
+	readonly triedPeerBucketCount?: number;
+	readonly triedPeerBucketSize?: number;
+	readonly maxReconnectTries?: number;
+	readonly secret: number;
+}
+
+interface TriedPeerInfo {
+	readonly peerInfo: P2PDiscoveredPeerInfo;
+	// tslint:disable-next-line:readonly-keyword
+	numOfConnectionFailures: number;
+	readonly dateAdded: Date;
+}
+
+export interface AddPeerOutcome {
+	readonly success: boolean;
+	readonly evicted: boolean;
+	readonly evictedPeer?: P2PPeerInfo;
+}
+
+export class TriedPeers {
+	private readonly _triedPeerMap: Map<number, Map<string, TriedPeerInfo>>;
+	private readonly _triedPeerBucketCount: number;
+	private readonly _triedPeerBucketSize: number;
+	private readonly _maxReconnectTries: number;
+	private readonly _secret: number;
+
+	public constructor({
+		triedPeerBucketCount,
+		maxReconnectTries,
+		secret,
+		triedPeerBucketSize,
+	}: TriedPeerConfig) {
+		this._triedPeerBucketCount = triedPeerBucketCount
+			? triedPeerBucketCount
+			: DEFAULT_TRIED_PEER_LIST_SIZE;
+		this._triedPeerBucketSize = triedPeerBucketSize
+			? triedPeerBucketSize
+			: DEFAULT_TRIED_PEER_BUCKET_SIZE;
+		this._maxReconnectTries = maxReconnectTries
+			? maxReconnectTries
+			: DEFAULT_MAX_RECONNECT_TRIES;
+		this._secret = secret;
+		// Initialize the Map with all the buckets
+		this._triedPeerMap = new Map();
+		[...Array(this._triedPeerBucketCount).keys()]
+			.map(x => x + 1)
+			.forEach(bucketId => {
+				this._triedPeerMap.set(bucketId, new Map<string, TriedPeerInfo>());
+			});
+	}
+
+	public get triedPeerConfig(): TriedPeerConfig {
+		return {
+			maxReconnectTries: this._maxReconnectTries,
+			triedPeerBucketSize: this._triedPeerBucketSize,
+			triedPeerBucketCount: this._triedPeerBucketCount,
+			secret: this._secret,
+		};
+	}
+
+	public get triedPeerMap(): Map<number, Map<string, TriedPeerInfo>> {
+		return this._triedPeerMap;
+	}
+
+	public getTriedPeersList(): ReadonlyArray<P2PDiscoveredPeerInfo> {
+		const peersListMap: P2PDiscoveredPeerInfo[] = [];
+
+		[...this._triedPeerMap.values()].forEach(peerMap => {
+			peerMap.forEach(peer => peersListMap.push(peer.peerInfo));
+		});
+
+		return peersListMap;
+	}
+
+	public getBucketId(ipAddress: string): number {
+		return getBucket({
+			secret: this._secret,
+			peerType: PEER_TYPE.TRIED_PEER,
+			targetAddress: ipAddress,
+		});
+	}
+
+	public findPeer(peerInfo: P2PPeerInfo): boolean {
```

Comment:
same naming issue as for newPeers above

Reply:
removed this function and using `getPeers` instead